pkg/tarfs/fs.go

```diff
@@ -491,7 +491,14 @@ func (m *memFS) writeHeader(name string, te tarEntry) error {
 	}
 
 	sameOrigin := got.pkg.Origin == want.pkg.Origin
-	replaces := got.pkg.Name == want.pkg.Replaces
+
+	replaces := false
+	for _, replace := range want.pkg.Replaces {
```

Comment:
Do we need to check both directions? It's valid if the package I'm installing replaces the one that exists, or if the package that's installed replaces the one I'm installing right?

Reply:
I think you're right but I think that's a bit different from the current behavior, so I'm going to merge this and we can fix that later too.